segments-explainer/SEGMENTS-EXPLAINER.md

```diff
@@ -0,0 +1,100 @@
+- [Window Placement API Explainer](https://github.com/webscreens/window-placement/blob/master/EXPLAINER.md) is useful in multi-window scenarios on multiple screen devices, but does not target scenarios in which the hosting application (i.e. browser) has a single window which spans multiple displays. In this case, the developer may not wish to open new windows - just hints to help lay out things properly and take advantage of the physical partitioning of the available layout space.
+ 
+Additionally, while not a solution in the same sense, a ["[css-media-queries] Foldables support and enablement"](https://github.com/w3c/csswg-drafts/issues/4141) issue discusses the problem space and outlines some details and touches upon outlined issues.
+
+## Proposal: Viewport Segments Property
+
+A summary of the concepts from the other proposals:
+* Display - the logical representation of an physical monitor.
+* Screen - the aggregate 2D space occupied by all the connected displays.
+
+We propose a new concept of Viewport Segments, which represent the regions of the window that reside on separate (adjacent) displays. Viewport Segment dimensions are expressed in CSS pixels and will be exposed via a JavaScript API that allows developers to enumerate segments where logically separate pieces of content can be placed. 
+
+This proposal is primarily aimed at reactive scenarios, where an application wants to take advantage of the fact that it spans multiple displays, by virtue of the user/window manager placing it in that state. It is not designed for scenarios of proactively placing content in a separate top-level browsing context on the various displays available (this would fall under the [Window Placement API](https://github.com/webscreens/window-placement/blob/master/EXPLAINER.md) or [Presentation API](https://w3c.github.io/presentation-api/)). Note that given the [Screen Enumeration API](https://github.com/webscreens/screen-enumeration/blob/master/EXPLAINER.md) and existing primitives on the Web, it is possible to write JavaScript code that intersects the rectangles of the Display and window, while taking into account devicePixelRatio in order to compute the interesting layout regions of a window spanned across displays. However this may not correctly handle corner cases of future device form factors, and thus this proposal tries to centralize access to "here are the interesting parts of the screen a developer can target or consider for presenting content" as a practical starting point.
+
+```
+interface VisualViewport : EventTarget {
+	readonly attribute FrozenArray<DOMRect> segments;
+}
+```
+
+The value returned from querying `visualViewport.segments` will be an array of DOMRects and based on the data returned for each "viewport segment", developers will be able to infer the number of hinges available as well as the hinge orientation. When `visualViewport.segments` is queried on a desktop, this will return `null` since there is only one viewport segment.
```

Comment:
instead of 'on a desktop' I think we should say when there is only a single segment, and mention it is not exposed in this case because this information would be redundant with other fields on VisualViewport (and also could lead to future compat issues on these devices if authors started using visualViewport.segments[0] to target single-screen devices)